src/core/net/netif.cpp

```diff
@@ -219,6 +217,7 @@ otError Netif::SubscribeAllRoutersMulticast(void)
     // Ensure that the `LinkLocalAll` was found on the list.
 
     OT_ASSERT(error == OT_ERROR_NONE);
+    OT_UNUSED_VARIABLE(error);
```

Comment:
Not in this PR, but if we define `OT_ASSERT(x)` to `((void)(x))` when it's **turned off**, we can get rid of this `OT_UNUSED_VARIABLE(error)`.

Reply:
@simonlingoogle I would like this (considered the same when changing `OT_ASSERT`) but realized that it may not work in all cases/toolchain. 

The  `(void)(x)` does not seem work on some tool-chains. That's why we have specific implementations of `OT_UNUSED_VARIABLE()` in `toolchain.h` which uses the address of the variable and I don't think we can apply the same trick to the `condition` check in `OT_ASSERT(condition)` (since it's not a variable).

We need a different approach/idea here :)

Reply:
Maybe something like:
```
#define OT_ASSERT(cond) \
    do                  \
    {                   \
        if (!(cond))    \
        {               \
        }               \
    } while (0)  
```

Reply:
I think your approach should be better. @abtink